### PR TITLE
processor(cdc): seperate errors of sub components

### PR DESCRIPTION
**Pull request by hicqu**
### What problem does this PR solve?

Issue Number: ref #8657 

### What is changed and how it works?

* Seperate errors of processor sub-components
  - sourcemanager
  - sinkmanager
  - redo
  - mounter
* Unify processor components into `Runnable`

Currently they are only sperated into different error channels, but still share same error handler. We will specify different error handler for them later in another PR.

#### Tests 

 - Unit test

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by nongfushanquan — COMMENTED** [review]
Please describe  the main idea of this pr to handle error

File: pkg/util/runnable.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+// r := createRunable()
+// ctx, cancel := context.WithCancel(context.Background())
+// go func() { handleError(r.Run(ctx)) }()
+// r.WaitForReady()
+// cancel()
+// r.Close()
+type Runnable interface {
+	// Run all sub goroutines and block the current one. If an error occurs
+	// in any sub goroutine, return it and cancel all others.
+	Run(ctx context.Context) error
```
Comment (CharlesCheung96) [inline pkg/util/runnable.go:32]: Whether the blocked Run releases resources before exiting? If so, is the Close method redundant?

**Review by sdojjy — APPROVED** [review]

**Review by CharlesCheung96 — APPROVED** [review]

**Pull request merged**